pytorch_lightning/core/lightning.py

```diff
@@ -1282,7 +1282,7 @@ def optimizer_step(self, epoch, batch_idx, optimizer, optimizer_idx,
         """
         if not isinstance(optimizer, LightningOptimizer):
```

Comment:
This is legacy code. It shouldn't be needed as we always provide a LightningOptimizer. Can you check ?

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/blob/release/1.2-dev/pytorch_lightning/trainer/training_loop.py#L504
I will remove it another PR.